auth-connector/src/main/java/com/sequenceiq/cloudbreak/auth/altus/service/AltusIAMService.java

```diff
@@ -15,33 +17,67 @@
 
     private static final Logger LOGGER = LoggerFactory.getLogger(AltusIAMService.class);
 
+    private final boolean useSharedAltusCredential;
+
+    private final String accessKey;
+
+    private final char[] secretKey;
+
     private final GrpcUmsClient umsClient;
 
-    public AltusIAMService(GrpcUmsClient umsClient) {
+    public AltusIAMService(GrpcUmsClient umsClient,
+            AltusDatabusConfiguration altusDatabusConfiguration) {
         this.umsClient = umsClient;
+        this.useSharedAltusCredential = altusDatabusConfiguration.isUseSharedAltusCredential();
+        this.accessKey = altusDatabusConfiguration.getSharedAccessKey();
+        this.secretKey = altusDatabusConfiguration.getSharedSecretKey();
     }
 
     /**
      * Generate machine user with access keys
      */
-    public Optional<AltusCredential> generateMachineUserWithAccessKey(String machineUserName, String actorCrn) {
+    public Optional<AltusCredential> generateMachineUserWithAccessKey(String machineUserName, String actorCrn, boolean useSharedCredential) {
+        if (useSharedAltusCredential) {
+            LOGGER.debug("Use shared altus credential is turned on for generating altus credential and access keys.");
+            if (areDatabusCredentialsFilled(useSharedCredential)) {
+                LOGGER.debug("Access and secret keys are set manually application wide for Databus, skip machine user and access key generation");
+                return Optional.of(new AltusCredential(accessKey, secretKey));
+            }
```

Comment:
Maybe adding an `else` clause with another log msg would be helpful for debugging. Alternatively, the 2 `if`s could be collapsed into a single one that checks `useSharedAltusCredential && areDatabusCredentialsFilled(useSharedCredential)`, similarly to `clearMachineUser()` below.